### PR TITLE
Add a red border when the form field is errored

### PR DESCRIPTION
**Pull request by Nicofuma**

### CODE REVIEW

**Comment by rpg600** [comment]
Instead of adding an error class each time a new field is added, you can use the form theming and override the form_widget or widget_attributes block.

**Comment by Nicofuma** [comment]
> Instead of adding an error class each time a new field is added, you can use the form theming and override the form_widget or widget_attributes block.

done

**Comment by Nicofuma** [comment]
rebased

File: src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig [hunk 1]
```diff
@@ -20,6 +20,30 @@
     {% endspaceless %}
 {% endblock form_errors %}
 
+{%- block widget_attributes -%}
+    id="{{ id }}" name="{{ full_name }}"
+    {%- if read_only and attr.readonly is not defined %} readonly="readonly"{% endif -%}
+    {%- if disabled %} disabled="disabled"{% endif -%}
+    {%- if required %} required="required"{% endif -%}
+
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig]: i think this code, can be optimized to something like
```
{%- block widget_attributes -%}
   {% set attr.class = "formError " ~ attr.class|default("") %}
{{ parent() }} #or something similar a with a use http://twig.sensiolabs.org/doc/tags/use.html
{%- endblock widget_attributes -%}
```

Comment (Nicofuma) [inline src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig]: you can't set a composed variable (`{% set attr.class = "something" %}` will fail)

Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig]:
```twig
{% set attr['class'] = "something" %}
```
??

Comment (Nicofuma) [inline src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig]: doesn't work either: `Unexpected token "punctuation" of value "[" ("end of statement block" expected)`

Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig]: Abort mission as the parent will not have the proper attr array in its context

Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig]: i think we should give a try to : 
``` 
#in the top of the file
{%  use 'form_div_layout.html.twig' with widget_attributes as base_widget_attributes %}

{%- block widget_attributes -%}
   {% set attr = attr|merge({"class" : "formError " ~ attr.class|default("")}) %}
{{ block('base_widget_attributes') }} 
{%- endblock widget_attributes -%}
```

Comment (ogizanagi) [inline src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig]: :+1: It actually works.

```diff
Index: src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig
<+>UTF-8
===================================================================
--- src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig	(revision f788dbca86afd9a8fa3edadda7361415ac66c449)
+++ src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig	(revision )
@@ -1,3 +1,5 @@
+{% use 'form_div_layout.html.twig' with widget_attributes as base_widget_attributes %}
+
 {% block form_row %}
     <div class="form_row">
         {{ form_widget(form) }}
@@ -21,27 +23,8 @@
 {% endblock form_errors %}
 
 {%- block widget_attributes -%}
-    id="{{ id }}" name="{{ full_name }}"
-    {%- if read_only and attr.readonly is not defined %} readonly="readonly"{% endif -%}
-    {%- if disabled %} disabled="disabled"{% endif -%}
-    {%- if required %} required="required"{% endif -%}
-
-    {%- for attrname, attrvalue in attr -%}
-        {{- " " -}}
-        {%- if attrname in ['placeholder', 'title'] -%}
-            {{- attrname }}="{{ attrvalue|trans({}, translation_domain) }}"
-        {%- elseif attrvalue is same as(true) -%}
-            {{- attrname }}="{{ attrname }}"
-        {%- elseif attrvalue is not same as(false) -%}
-            {%- if not form.vars.valid and attrname == "class" -%}
-                {%- set attrvalue = (attrvalue ~ " formError") -%}
-            {%- endif -%}
-
-            {{- attrname }}="{{ attrvalue }}"
-        {%- endif -%}
-    {%- endfor -%}
-
-    {%- if not form.vars.valid and not attr.class is defined %} class="formError"{%- endif -%}
+    {% set attr = attr|merge({"class" : "formError " ~ attr.class|default("")}) %}
+    {{- block('base_widget_attributes') -}}
 {%- endblock widget_attributes -%}
 
 {% block _job_widget %}
```

Comment (Nicofuma) [inline src/SensioLabs/JobBoardBundle/Resources/views/Form/forms.html.twig]: updated

**Comment by salahm** [comment]
:+1:

**Comment by ogizanagi** [comment]
:+1:

**Pull request merged**